### PR TITLE
update ShenyuGrpcClient

### PR DESCRIPTION
**Pull request by weihubeats**
https://github.com/apache/incubator-shenyu/issues/3485

### CODE REVIEW

File: shenyu-client/shenyu-client-grpc/src/main/java/org/apache/shenyu/client/grpc/GrpcClientBeanPostProcessor.java [hunk 1]
```diff
@@ -125,7 +125,7 @@ private void handler(final Object serviceBean) {
             LOG.error(String.format("grpc SERVICE_NAME can not found: %s", classes));
             return;
         }
-        ShenyuGrpcClient grpcClassAnnotation = AnnotationUtils.findAnnotation(clazz, ShenyuGrpcClient.class);
+        ShenyuGrpcClient grpcClassAnnotation = AnnotatedElementUtils.findMergedAnnotation(clazz, ShenyuGrpcClient.class);
```
[the remark this answers is not in the document]
Comment (dragon-zhang) [reply shenyu-client/shenyu-client-grpc/src/main/java/org/apache/shenyu/client/grpc/GrpcClientBeanPostProcessor.java:128]: `AnnotationUtils#findAnnotation` can be compatible with AOP, but `AnnotatedElementUtils#findMergedAnnotation` will not.

[the remark this answers is not in the document]
Comment (weihubeats) [reply shenyu-client/shenyu-client-grpc/src/main/java/org/apache/shenyu/client/grpc/GrpcClientBeanPostProcessor.java]: If the annotation on the method is empty, get the annotation on the class？

[the remark this answers is not in the document]
Comment (weihubeats) [reply shenyu-client/shenyu-client-grpc/src/main/java/org/apache/shenyu/client/grpc/GrpcClientBeanPostProcessor.java]: I've dealt with it, take a look

[the remark this answers is not in the document]
Comment (dragon-zhang) [reply shenyu-client/shenyu-client-grpc/src/main/java/org/apache/shenyu/client/grpc/GrpcClientBeanPostProcessor.java]: > If the annotation on the method is empty, get the annotation on the class？

no

[the remark this answers is not in the document]
Comment (dragon-zhang) [reply shenyu-client/shenyu-client-grpc/src/main/java/org/apache/shenyu/client/grpc/GrpcClientBeanPostProcessor.java]: > I've dealt with it, take a look

you mean this ?
```java
        Class<?> clazz;
        try {
            clazz = serviceBean.getClass();
        } catch (Exception e) {
            LOG.error("failed to get grpc target class", e);
            return;
        }
        Class<?> parent = clazz.getSuperclass();
        Class<?> classes = parent.getDeclaringClass();
        String packageName;
        try {
            String serviceName = ShenyuClientConstants.SERVICE_NAME;
            Field field = classes.getField(serviceName);
            field.setAccessible(true);
            packageName = field.get(null).toString();
        } catch (Exception e) {
            LOG.error(String.format("SERVICE_NAME field not found: %s", classes), e);
            return;
        }
```

File: shenyu-client/shenyu-client-grpc/src/main/java/org/apache/shenyu/client/grpc/GrpcClientBeanPostProcessor.java [hunk 2]
```diff
@@ -125,21 +125,22 @@ private void handler(final Object serviceBean) {
...
             }
             return;
         }
         final Method[] methods = ReflectionUtils.getUniqueDeclaredMethods(clazz);
         for (Method method : methods) {
-            ShenyuGrpcClient grpcClient = method.getAnnotation(ShenyuGrpcClient.class);
-            if (Objects.nonNull(grpcClient)) {
-                publisher.publishEvent(buildMetaDataDTO(packageName, grpcClient, method, basePath));
+            ShenyuGrpcClient methodShenyuClient = AnnotatedElementUtils.findMergedAnnotation(method, ShenyuGrpcClient.class);
+            methodShenyuClient = Objects.isNull(methodShenyuClient) ? beanShenyuClient : methodShenyuClient;
```
[the remark this answers is not in the document]
Comment (weihubeats) [reply shenyu-client/shenyu-client-grpc/src/main/java/org/apache/shenyu/client/grpc/GrpcClientBeanPostProcessor.java]: I deleted the get class annotation, sick get aop class

[the remark this answers is not in the document]
Comment (dragon-zhang) [reply shenyu-client/shenyu-client-grpc/src/main/java/org/apache/shenyu/client/grpc/GrpcClientBeanPostProcessor.java]: > I deleted the get class annotation, sick get aop class

Nice work, this PR will be merged after CI/CD passed.

**Review by dragon-zhang — APPROVED** [review]

**Comment by dragon-zhang** [comment]
@weihubeats please add wechat Z18179469818

**Pull request merged**